source/agora/node/Ledger.d

```diff
@@ -227,31 +254,43 @@ public class Ledger
 
     public Hash[] getMerklePath (ulong block_height, Hash hash) @safe
     {
-        if (block_height >= this.ledger.length)
+        if (this.block_height < block_height)
             return null;
 
-        const(Block)* block = &this.ledger[block_height];
-
-        size_t index = block.findHashIndex(hash);
-        if (index < block.txs.length)
-            return block.getMerklePath(index);
+        Block block;
+        if (this.storage.readBlock(block, block_height))
+        {
+            size_t index = block.findHashIndex(hash);
+            if (index < block.txs.length)
+                return block.getMerklePath(index);
+            else
+                return null;
+        }
         else
-            return null;
+        {
+            assert(0);
+        }
     }
 }
 
 ///
 unittest
 {
+    import agora.common.BlockStorage;
     import agora.common.crypto.Key;
     import agora.common.Data;
     import agora.common.Hash;
 
+    string path = buildPath(getcwd, ".cache");
+    if (path.exists)
+        rmdirRecurse(path);
```

Comment:
We should not do IO in unittests. Do we have a way to have in memory BlockStorage @AndrejMitrovic ?

Reply:
I already made a comment about this. https://github.com/bpfkorea/agora/pull/162#discussion_r312346572